File: Stack/Valid_Parantheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        charDict = {']':'[', '}': '{', ')':'('}

        if len(s) < 2:
            return False

        for char in s:
            if char == '{' or char == '[' or char == '(':
                stack.append(char)
            else:
                if len(stack) < 1:
                    return False
                elif stack.pop() != charDict[char]:
                    return False
        
        if len(stack) > 0:
            return False
        else:
            return True

File: Stack/Valid_Parantheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];

    for (i = 0; i < s.length; i++) {
        if (s[i] === '[' || s[i] === '{' || s[i] === '(') {
            stack.push(s[i]);
        }
        else if (s[i] === ']') {
            if (stack[stack.length - 1] === '[') {
                stack.pop();
            }
            else {
                return false;
            }
        }
        else if (s[i] === '}') {
            if (stack[stack.length - 1] === '{') {
                stack.pop();
            }
            else {
                return false;
            }
        }
        else if (s[i] === ')') {
            if (stack[stack.length - 1] === '(') {
                stack.pop();
            }
            else {
                return false;
            }
        }
    }

    if (stack.length === 0) {
        return true;
    }
    else {return false;}
};